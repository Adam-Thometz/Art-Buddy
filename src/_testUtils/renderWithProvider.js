import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;