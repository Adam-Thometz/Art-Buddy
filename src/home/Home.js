import React from "react";
import useLocalStorage from "_hooks/useLocalStorage";

import Welcome from "./welcome/Welcome";
import Menu from "./menu/Menu";

const Home = () => {
  const [hasVisited, setHasVisited] = useLocalStorage('visited');
  return !hasVisited ? <Welcome setHasVisited={setHasVisited} /> : <Menu />
};

export default Home;