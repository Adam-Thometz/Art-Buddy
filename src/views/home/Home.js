import useVisited from "hooks/visited/useVisited";

import Welcome from "./welcome/Welcome";
import Menu from "./menu/Menu";

const Home = () => {
  const [hasVisited, setHasVisited] = useVisited();
  return !hasVisited ? <Welcome setHasVisited={setHasVisited} /> : <Menu />;
};

export default Home;
