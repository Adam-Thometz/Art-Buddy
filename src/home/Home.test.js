import React from "react";
import renderWithProvider from '_testUtils/renderWithProvider';
import Home from './Home';

describe('Home component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<Home />);
  });

  it('matches the snapshot of the landing page displaying the games page', () => {
    window.localStorage.setItem('visited', true);
    const { asFragment } = renderWithProvider(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('matches the snapshot of the landing page displaying the welcome page', () => {
    window.localStorage.setItem('visited', false);
    const { asFragment } = renderWithProvider(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});