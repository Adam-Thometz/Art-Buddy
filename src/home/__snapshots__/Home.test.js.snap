// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches the snapshot of the landing page displaying the games page 1`] = `
<DocumentFragment>
  <main
    class="Menu"
  >
    <header
      class="WindowNavbar-main"
    >
      <span
        class="WindowNavbar-button underlined"
        id="games"
      >
        GAMES
      </span>
      <span
        class="WindowNavbar-button"
        id="tools"
      >
        TOOLS
      </span>
    </header>
    <div
      class="Menu-options"
    >
      <button
        class="Button regular"
        style="border-color: #89eb0a;"
      >
        <img
          alt=""
          src="file"
        />
        ART GAMES
      </button>
      <button
        class="Button regular"
        style="border-color: #ffe81a;"
      >
        <img
          alt=""
          src="file"
        />
        MUSIC GAMES
      </button>
      <button
        class="Button regular"
        style="border-color: #fd4d3a;"
      >
        <img
          alt=""
          src="file"
        />
        ALL GAMES
      </button>
      <button
        class="Button regular"
        disabled=""
        style="border-color: #4d4f5e; background-color: grey;"
      >
        <img
          alt=""
          src="file"
        />
        IEP GOALS
      </button>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Home component matches the snapshot of the landing page displaying the welcome page 1`] = `
<DocumentFragment>
  <section
    class="Welcome"
  >
    <header
      class="WindowNavbar-games"
    >
      <h1
        class="WindowNavbar-label"
      >
        WELCOME TO ART BUDDY!
      </h1>
      <img
        alt="Help"
        class="Help"
        src="file"
      />
    </header>
    <article
      class="Welcome-description"
    >
      <p>
        ART BUDDY is a collection of fun and interactive fine arts games designed specifically with academic goals in mind
      </p>
      <p>
        Click below to get started
      </p>
    </article>
    <button
      class="Button regular"
      style="border-color: #89eb0a;"
    >
      LET'S GO!
    </button>
  </section>
</DocumentFragment>
`;
